packages/rabbitmq/src/rabbitmq.module.ts

```diff
@@ -81,20 +80,28 @@ export class RabbitMQModule
     @Inject(RABBIT_CONFIG_TOKEN) config: RabbitMQConfig
   ) {
     super();
-    this.logger = config.logger || new Logger(RabbitMQModule.name);
+    this.logger = config?.logger || new Logger(RabbitMQModule.name);
   }
 
-  static async AmqpConnectionFactory(config: RabbitMQConfig) {
+  static async AmqpConnectionFactory(
+    config: RabbitMQConfig
+  ): Promise<AmqpConnection | undefined> {
+    const logger = config?.logger || new Logger(RabbitMQModule.name);
+    if (config == undefined) {
```

Comment:
possibly just do `if (!config)`, you don't need to explicitly check for undefined